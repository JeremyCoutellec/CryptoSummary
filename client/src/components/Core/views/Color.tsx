import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import clsx from 'clsx';

// Material UI
import { makeStyles } from '@mui/styles';
import Tooltip from '@mui/material/Tooltip';

const useStyles = makeStyles((theme) => ({
    colorDiv: {
        padding: theme.spacing(1),
        cursor: 'pointer',
        display: 'inline-block',
        border: 'solid',
        float: 'right',
    },
    colorDivDense: {
        lineHeight: 'initial',
        padding: theme.spacing(0.5),
    },
}));

const Color = ({ value, dense = false, ...rest }) => {
    const classes = useStyles();
    const { t } = useTranslation();
    const [copied, setCopied] = useState(false);

    const copyValue = () => {
        navigator.clipboard.writeText(value);
        setCopied(true);
        setTimeout(function () {
            setCopied(false);
        }, 2000);
    };

    return value ? (
        <Tooltip
            title={t('common.copied')}
            open={copied}
            placement="top-end"
            leaveDelay={200}
            disableFocusListener
            disableHoverListener
            disableTouchListener
        >
            <div className={clsx(classes.colorDiv, dense && classes.colorDivDense)} onClick={copyValue} style={{ borderColor: value }} {...rest}>
                {value}
            </div>
        </Tooltip>
    ) : (
        <div />
    );
};

export default Color;
