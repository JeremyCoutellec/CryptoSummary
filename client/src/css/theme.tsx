import { Theme, createTheme } from '@mui/material/styles';
import { frFR as dataGridFR, enUS as dataGridEN } from '@mui/x-data-grid';
import { frFR, enUS } from '@mui/material/locale';

declare module '@mui/styles/defaultTheme' {
    // eslint-disable-next-line @typescript-eslint/no-empty-interface
    interface DefaultTheme extends Theme {}
}

// A custom theme for this app
let theme = createTheme(
    {
        components: {
            MuiTextField: {
                defaultProps: {
                    variant: 'outlined',
                },
            },
            MuiLink: {
                defaultProps: {
                    underline: 'always',
                },
            },
        },
        palette: {
            primary: {
                dark: '#183754',
                main: '#424B62',
                light: '#6DCDEE',
            },
            secondary: {
                dark: '#F37073',
                main: '#BEBFBB',
                light: '#F3F3F4',
            },
            error: {
                main: '#E30613',
                light: '#e30613a1',
            },
            warning: {
                main: '#F39207',
                light: '#f39207a1',
            },
            success: {
                main: '#4BBF9B',
                light: '#6fcbafba',
            },
            background: {
                default: '#fff',
            },
        },
        typography: {
            fontFamily: 'Roboto',
            h1: {
                fontSize: '2.5rem',
                fontWeight: 700,
                lineHeight: 1.25,
            },
            h2: {
                fontSize: '2rem',
                fontWeight: 700,
                lineHeight: 1.3333,
            },
            h3: {
                fontSize: '1.5rem',
                fontWeight: 700,
                lineHeight: 1.5,
            },
            h4: {
                fontSize: '1.25rem',
                fontWeight: 700,
                lineHeight: 1.5,
            },
            h5: {
                fontSize: '1.125rem',
                fontWeight: 700,
                lineHeight: 1.5,
            },
            h6: {
                fontSize: '1.0625rem',
                fontWeight: 700,
                lineHeight: 1.5,
            },
            subtitle1: {
                fontSize: '1rem',
                fontWeight: 600,
                lineHeight: 1.55555,
            },
            subtitle2: {
                fontSize: '0.875rem',
                fontWeight: 600,
                lineHeight: 1.57143,
            },
            body1: {
                fontSize: '1rem',
                fontWeight: 400,
                lineHeight: 1.5,
            },
            body2: {
                fontSize: '0.875rem',
                fontWeight: 400,
                lineHeight: 1.57143,
            },
            caption: {
                fontSize: '0.75rem',
                fontWeight: 400,
                lineHeight: 1.5,
            },
            button: {
                fontSize: '0.875rem',
                fontWeight: 700,
                lineHeight: 1.71429,
            },
            overline: {
                fontSize: '0.75rem',
                fontWeight: 700,
                lineHeight: 1.5,
            },
        },
    },
    localStorage.getItem('locale') === 'en' ? dataGridEN : dataGridFR,
    localStorage.getItem('locale') === 'en' ? enUS : frFR
);

export default theme;
