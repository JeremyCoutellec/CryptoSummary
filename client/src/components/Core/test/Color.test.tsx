import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/material/styles';
import theme from '../../../css/theme';

import Color from '../views/Color';

jest.mock('react-i18next', () => ({
    useTranslation: () => ({ t: (key) => key }),
}));

test('Color rendering', () => {
    const propsColor = {
        value: '#ffffff',
        dense: true,
    };
    const component = renderer.create(
        <ThemeProvider theme={theme}>
            <Color {...propsColor} />
        </ThemeProvider>
    );
    const color = component.toJSON();
    expect(color).toMatchSnapshot();
});
