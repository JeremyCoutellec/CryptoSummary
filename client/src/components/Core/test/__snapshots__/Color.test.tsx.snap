// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color rendering 1`] = `
<div
  aria-label="common.copied"
  aria-labelledby={null}
  className="makeStyles-colorDiv-1 makeStyles-colorDivDense-2"
  data-mui-internal-clone-element={true}
  onClick={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "borderColor": "#ffffff",
    }
  }
>
  #ffffff
</div>
`;
